### ISSUE TITLE
Wrong calculation for file size of gzipped files

### ISSUE THREAD

**Issue by shaman-apprentice** [MaibornWolff/codecharta#3111]
# Bug

## Expected Behavior

Given a gzipped file
When uploading it
Then the calculated file size is correct

## Actual Behavior
The calculated file size is always 13

## Steps to Reproduce the Problem
1. upload a gz, e.g. *app/codeCharta/assets/output.cc.json.gz*

## Technical analysis
Calculated file size comes from `response.body.toString().length` . `response.body` is a blob. Therefore `response.body.toString()` is always the string `"[object Blob]"` and its length is `13`

## Specifications
- is released in [online-demo](https://maibornwolff.github.io/codecharta/visualization/app/index.html?file=codecharta.cc.json.gz&file=codecharta_analysis.cc.json.gz): yes
- CodeCharta Version: 1.109.1

**Comment by ce-bo** [comment]
We need the correct file size to handle the visualization of big maps and to avoid crashes of CC due to performance issues. Hence, it should be fixed.

In more detail: We try to scale down the resolution of the rendered city map according to the map size to prevent rendered maps with a resolution, for instance, of 50.000 x 50.000 pixels.

**Comment by BridgeAR** [comment]
@ce-bo we probably do not need the file size for that as it is actually not representing the actual size of the map (long metric names and lots of metrics with longer values would not be an issue while vewer metrics with short names might be an issue if the map is big and the file size is identical.
The implementation of @RomanenkoVladimir takes an absolute value for the calculated size and makes sure the map does not become any bigger. That should suffice in this case.
Is the file size maybe also used for hash value? We could in fact add that to reduce computation time (no need to calculate a hash in case a file size is not identical).

**Comment by ce-bo** [comment]
I was not aware, that this will change in the implementation of @RomanenkoVladimir, nice! So, we do not need it anymore.
Your suggestion to check the file size before calculating the map hash is fine though.
@RomanenkoVladimir did you already remove the file size calculation in #3077?

**Comment by RomanenkoVladimir** [comment]
I am not sure we can remove that, my issue only addresses TreeMap, but what about StreetView maps?

**Comment by ce-bo** [comment]
AFAIK, we do not use the scaling for StreetMaps.

**Comment by RomanenkoVladimir** [comment]
but we call the mapResoultionScaling algorithm and flatly removing it leads to crashes in firefox

**Comment by RomanenkoVladimir** [comment]
I think we need to investigate this once #3077 is merged

**Comment by BridgeAR** [comment]
Seems like this is unblocked now? :tada:

**Comment by ce-bo** [comment]
> Seems like this is unblocked now? 🎉

Indeed. I would like to check, if we can just remove it or fix it to use it as a pre-condition for calculating the md5 hashes (map checksum). I increased the priority accordingly.
